modules/utxo-lib/test/integration_local_rpc/parse.ts

```diff
@@ -89,15 +97,27 @@ function runTestParse(protocol: Protocol, txType: FixtureTxType, scriptType: Scr
       return prevOutput;
     }
 
-    function getPrevOutputValue(input: InputLookup) {
-      return getPrevOutput(input).value * 1e8;
+    function getValueScaled(value: number, scaleFactor = 1e8): TNumber {
+      if (amountType === 'number') {
+        return (Number(value) * scaleFactor) as TNumber;
+      } else {
+        const integerPart = Math.floor(value);
+        let decimalPart = value - integerPart;
+        decimalPart = Math.round(decimalPart * scaleFactor); // scale and fix floating point
+
+        return (BigInt(integerPart) * BigInt(scaleFactor) + BigInt(decimalPart)) as TNumber;
```

Comment:
seeing this a second time now - this should be a separate func with own tests

Reply:
created new function in `transaction_util`